arm/Microsoft.OperationsManagement/solutions/deploy.bicep

```diff
@@ -0,0 +1,61 @@
+@description('Required. Name of the solution.')
+param name string
+
+@description('Required. Name of the Log Analytics workspace where the solution will be deployed/enabled.')
+param logAnalyticsWorkspaceName string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. The product of the deployed solution. For Microsoft published gallery solution it should be OMSGallery. This is case sensitive.')
+param product string = 'OMSGallery'
+
+@description('Optional. The publisher name of the deployed solution. For gallery solution, it is Microsoft.')
+param publisher string = 'Microsoft'
+
+@description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
+param enableDefaultTelemetry bool = true
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' existing = {
+  name: logAnalyticsWorkspaceName
+}
+
+var solutionName = '${name}(${logAnalyticsWorkspace.name})'
```

Comment:
Can we rely on this, or should it be somehow conditional? Right now the user cannot choose the name - but that may be correct. Would just like to hear your thoughts

Reply:
Including reply to other related comments here:

From ARR https://docs.microsoft.com/en-us/azure/templates/microsoft.operationsmanagement/solutions?tabs=bicep
> - **name**	name of the solution to be created. For Microsoft published solution it should be in the format of solutionType(workspaceName). For third party solution, it can be anything.
> - **product**	name of the solution to enabled/add. For Microsoft published gallery solution it should be in the format of OMSGallery/{solutionType}.

So yes, third party solutions are a thing and the current module is not supporting them. Would something like this work maybe?

```
var solutionName = product == 'OMSGallery' ? '${name}(${logAnalyticsWorkspace.name})' : name
var solutionProduct = product == 'OMSGallery' ? 'OMSGallery/${name} : name
```

Will still be challenging to describe properly though 🌪️

Reply:
check